packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts

```diff
@@ -127,3 +127,11 @@ async function getLatestBlockNumber(provider: HttpProvider) {
   const latestBlock = BigInt(latestBlockString);
   return latestBlock;
 }
+
+export function getLastSafeBlock(
+  latestBlock: bigint,
+  maxReorg: bigint
+): bigint {
+  // Design choice: if latestBlock - maxReorg gives a negative number than the newest block will be used
```

Comment:
Let's explain that we do this because the forked network is almost surely a devnet

Reply:
Done